controllers/bpfman-agent/common.go

```diff
@@ -232,7 +232,7 @@ func (r *ReconcilerCommon) reconcileBpfProgram(ctx context.Context,
 				r.Logger.V(1).Info("bpf program is in wrong state, unloading and reloading", "reason", reasons, "bpfProgram Name", bpfProgram.Name, "bpf program ID", id)
 				if err := bpfmanagentinternal.UnloadBpfmanProgram(ctx, r.BpfmanClient, *id); err != nil {
 					r.Logger.Error(err, "Failed to unload BPF Program")
-					return bpfmaniov1alpha1.BpfProgCondNotUnloaded, nil
+					return bpfmaniov1alpha1.BpfProgCondNotUnloaded, err
```

Comment:
It seems reasonable to return an error here.  However, as the code stands, I think we will only try this one more time (unless a change to another program kicks off the reconciler again.)  If it fails the second time, we'll stop.

Reply:
We've talked about this being an issue, so I wanted to flag it for discussion.  Maybe we should keep trying, or at least try a few times.  We could count how many times it has failed in the CRD.  Maybe do some type of retry with a back off so we keep trying every once in a while.  If we could get a failure reason from bpfman, maybe we can determine whether it's a permanent or possibly transient failure to guide whether/when to do a retry.

Reply:
Ok.  I think these changes are all good.  I also think we should evaluate whether we should have retries when the bpfman calls fail, and how to do it.